include/deal.II/grid/tria.h

```diff
@@ -511,6 +513,167 @@ namespace internal
 /*------------------------------------------------------------------------*/
 
 
+namespace construction
+{
+  /**
+   * Configuration flags for fully distributed Triangulations.
+   * Settings can be combined using bitwise OR.
+   *
+   * @author Peter Munch, 2019
+   */
+  enum Settings
+  {
+    /**
+     * Default settings, other options are disabled.
+     */
+    default_setting = 0x0,
+    /**
+     * This flags needs to be set to use the geometric multigrid
+     * functionality. This option requires additional computation and
+     * communication.
+     */
+    construct_multigrid_hierarchy = 0x1
+  };
+
+  /**
+   * Information needed for each locally relevant cell, stored in
+   * ConstructionData and used during construction of a
+   * parallel::fullydistributed::Triangulation. This struct stores
+   * the cell id, the subdomain_id and the level_subdomain_id as well as
+   * information related to manifold_id and boundary_id.
+   *
+   * @author Peter Munch, 2019
+   */
+  template <int dim>
+  struct CellData
```

Comment:
Could this class be derived from `::CellData`? You'd then only have to implement whatever *else* you need for the current operation.

Reply:
I am not really in favor of this. The only thins which is common `material_id` and `manifold_id`. `::CellData::vertices` is not common. However, we can change this in the future depending on how @luca-heltai intends to implement his features.